src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -1220,6 +1220,9 @@ internal virtual bool TryApplyChanges(Solution newSolution, IProgressTracker pro
                     progressTracker.ItemCompleted();
                 }
 
+                // changes in mapped files outside the workspace (may span multiple projects)
+                this.ApplyMappedFileChanges(solutionChanges);
```

Comment:
Why is this necessary as opposed to just overriding TryApplyChanges and doing this after the call to base.TryApplyChanges?